projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kerberos_ticket.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+
+class KerberosTicket:
+    """
+    Adapted from http://python-notes.curiousefficiency.org/en/latest/python_kerberos.html
+    """
+
+    def __init__(self, service):
+        import kerberos
+
+        __, krb_context = kerberos.authGSSClientInit(service)
+        kerberos.authGSSClientStep(krb_context, "")
+        self._krb_context = krb_context
+        self.auth_header = "Negotiate " + kerberos.authGSSClientResponse(krb_context)
+
+    def verify_response(self, auth_header):
+        """
+        Optionally use this method to verify response in case the server is not trusted fully.
+
+        :param auth_header:
+        """
+        # Handle comma-separated lists of authentication fields
+        for field in auth_header.split(","):
+            kind, __, details = field.strip().partition(" ")
+            if kind.lower() == "negotiate":
+                auth_details = details.strip()
+                break
+        else:
+            raise ValueError("Negotiate not found in %s" % auth_header)
+        # Finish the Kerberos handshake
+        krb_context = self._krb_context
+        if krb_context is None:
+            raise RuntimeError("Ticket already used for verification")
+        self._krb_context = None
+        import kerberos
```

Comment:
Isn't kerberos imported in the `__init__`, why do it here as well?

Reply:
The import is scoped to a function. So the import  in `__init__` is only visible in `__init__`method. 
You can try it out yourself: 

```
def bar():
   import kerberos
   print(dir(kerberos))

bar() # will not fail
print(dir(kerberos)) # fails
```

Reply:
Oopsie, correct. My bad.

Reply:
In any case why don't we import it at the top of the file as is standard?